en/docs/references/extend/federation/write-a-custom-oauth2-federated-authenticator.md

```diff
@@ -0,0 +1,84 @@
+|[public boolean canHandle(HttpServletRequest request)](https://github.com/wso2-extensions/identity-outbound-auth-oauth2/blob/75745516bb81f3b93b47075da720cd2fbc194217/components/org.wso2.carbon.identity.application.authenticator.oauth2/src/main/java/org.wso2.carbon.identity.application.authenticator.oauth2/Oauth2GenericAuthenticator.java#L470-L473)|Specifies whether this authenticator can handle the authentication response.|
+|[protected void initiateAuthenticationRequest(HttpServletRequest request,HttpServletResponse response, AuthenticationContext context)](https://github.com/wso2-extensions/identity-outbound-auth-oauth2/blob/75745516bb81f3b93b47075da720cd2fbc194217/components/org.wso2.carbon.identity.application.authenticator.oauth2/src/main/java/org.wso2.carbon.identity.application.authenticator.oauth2/Oauth2GenericAuthenticator.java#L76-L113)|Redirects the user to the login page in order to authenticate and in this sample, the user is redirected to the login page of the Kakao.|
+|[protected void processAuthenticationResponse(HttpServletRequest request,HttpServletResponse response, AuthenticationContext context)](https://github.com/wso2-extensions/identity-outbound-auth-oauth2/blob/75745516bb81f3b93b47075da720cd2fbc194217/components/org.wso2.carbon.identity.application.authenticator.oauth2/src/main/java/org.wso2.carbon.identity.application.authenticator.oauth2/Oauth2GenericAuthenticator.java#L116-L145)|Implements the logic of the custom federated authenticator.|
+|[protected void buildClaims(AuthenticationContext context, String userInfoString)](https://github.com/wso2-extensions/identity-outbound-auth-oauth2/blob/75745516bb81f3b93b47075da720cd2fbc194217/components/org.wso2.carbon.identity.application.authenticator.oauth2/src/main/java/org.wso2.carbon.identity.application.authenticator.oauth2/Oauth2GenericAuthenticator.java#L147-L183)|Implements the logic for extracting the subject from claims and for other claim mappings.|
+
+---
+
+## Deploy the custom federated authenticator in WSO2 IS
+
+1. Once the implementation is done, navigate to the root of your project and run the following command to compile the
+   service.
+    ``` xml
+    mvn clean install
+    ```
+2. Copy the compiled `jar` file inside `/target` and paste it to the `<IS_HOME>/repository/components/dropins`.
+
+---
+
+## Configure custom federated authenticator
+
+1. To configure the custom authenticator, click the Add button under Identity Providers and add an IDP name to register the new IDP.
+2. Click Federated Authenticators and expand `OAuth2 Custom Authenticator Configuration`. Configure it as follows.
+
+    - Select the **Enable** and **Default** check boxes.
+    - **Client Id** - The client ID value generated by the external OAuth2 IdP.
+    - **Client Secret** - The client secret value generated by the external OAuth2 IdP.
+    - **Callback Url** - https://localhost:9443/commonauth
+    - **Authorization Endpoint URL** - https://kauth.kakao.com/oauth/authorize
```

Comment:
do we need to configure them? These needs to go inside the specific authenticator while deploying sicne these are authenticator specific endpoints

Reply:
Do we have such behavior in Kakao also? Do they need to explicitly configure the endpoint?